From 823ae1b194c52311c8b91ad02d12b74edb7356e5 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Fri, 22 May 2020 10:34:53 +0800
Subject: [PATCH 1391/1851] usb: dwc3: use PROPERTY_ENTRY_BOOL to initialize
 property entries

commit 780f85e888e78049d15aa7425651c3e57c9efd60 from
https://github.com/Xilinx/linux-xlnx.git

According to commit 5eb5afb07853d6e9 ("usb: dwc3: use proper initializers
for property entries"), property entries should be initialized by proper
initializer macros. If not, the elements "is_inline" and "length" in
structure property_entry will be zeros, and the function
"property_entry_copy_data" will return -ENODATA failure. This will result
that dwc3_host_init fails.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/dwc3/host.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/host.c b/drivers/usb/dwc3/host.c
index 7e4829f29eec..40f4db486bce 100644
--- a/drivers/usb/dwc3/host.c
+++ b/drivers/usb/dwc3/host.c
@@ -103,7 +103,7 @@ int dwc3_host_init(struct dwc3 *dwc)
 
 	if (device_property_read_bool(&dwc3_pdev->dev,
 					"snps,xhci-stream-quirk"))
-		props[prop_idx++].name = "xhci-stream-quirk";
+		props[prop_idx++] = PROPERTY_ENTRY_BOOL("xhci-stream-quirk");
 
 	/**
 	 * WORKAROUND: dwc3 revisions <=3.00a have a limitation
-- 
2.31.1

